File: simple-server.js
const http = require('http');
const fs = require('fs');
const path = require('path');
const url = require('url');

// Check if Node.js is available
console.log('Node.js version:', process.version);

// Define the port
const PORT = 8000;

// Look for WordPress directories
const wpDirs = [
    'C:/Users/tjd20.LAPTOP-PCMC2SUO/local-wordpress',
    'C:/Users/tjd20.LAPTOP-PCMC2SUO/wordpress',
    'C:/Users/tjd20.LAPTOP-PCMC2SUO/wordpress-local-test',
    '.'
];

let wpDir = '.';
for (const directory of wpDirs) {
    if (fs.existsSync(directory)) {
        // Check if it contains WordPress files
        if (fs.existsSync(path.join(directory, 'wp-config.php')) || 
            fs.existsSync(path.join(directory, 'wp-content')) || 
            fs.existsSync(path.join(directory, 'wp-includes'))) {
            wpDir = directory;
            break;
        }
    }
}

console.log(`Serving files from: ${wpDir}`);
process.chdir(wpDir);

// MIME types
const mimeTypes = {
    '.html': 'text/html',
    '.php': 'text/html', // We'll serve PHP files as HTML for simplicity
    '.css': 'text/css',
    '.js': 'text/javascript',
    '.json': 'application/json',
    '.png': 'image/png',
    '.jpg': 'image/jpeg',
    '.gif': 'image/gif',
    '.svg': 'image/svg+xml',
    '.ico': 'image/x-icon'
};

// Create the server
const server = http.createServer((req, res) => {
    const parsedUrl = url.parse(req.url);
    let pathname = `.${parsedUrl.pathname}`;
    
    // Handle directory requests
    if (pathname.endsWith('/')) {
        pathname += 'index.html';
    }
    
    // Handle PHP files (serve as static for testing)
    if (pathname.endsWith('.php')) {
        // For WordPress, we'll try to serve index.html instead
        pathname = pathname.replace('.php', '.html');
        if (!fs.existsSync(pathname)) {
            pathname = pathname.replace('.html', '/index.html');
        }
    }
    
    const ext = path.parse(pathname).ext;
    const map = mimeTypes[ext] || 'text/plain';
    
    fs.exists(pathname, (exist) => {
        if (!exist) {
            // File not found, send 404
            res.statusCode = 404;
            res.end(`File ${pathname} not found!`);
            return;
        }
        
        // If is a directory, then look for index.html
        if (fs.statSync(pathname).isDirectory()) {
            pathname += '/index.html';
        }
        
        // Read file from file system
        fs.readFile(pathname, (err, data) => {
            if (err) {
                res.statusCode = 500;
                res.end(`Error getting the file: ${err}.`);
            } else {
                // If the file is found, set Content-Type and send data
                res.setHeader('Content-type', map);
                res.end(data);
            }
        });
    });
});

// Start the server
server.listen(PORT, () => {
    console.log(`Server running at http://localhost:${PORT}/`);
    console.log('Press Ctrl+C to stop the server');
    
    // Try to open the browser
    try {
        const { exec } = require('child_process');
        exec(`start http://localhost:${PORT}`, (error) => {
            if (error) {
                console.log('Could not open browser automatically');
            }
        });
    } catch (e) {
        console.log('Could not open browser automatically');
    }
});

// Handle graceful shutdown
process.on('SIGINT', () => {
    console.log('\nServer stopped.');
    process.exit(0);
});

File: simple-server.py
#!/usr/bin/env python3
"""
Simple HTTP Server for testing WordPress files
This is a basic server for testing purposes only.
For full WordPress functionality, use XAMPP with Apache and MySQL.
"""

import http.server
import socketserver
import os
import webbrowser
import threading
import time

# Check if we're in the right directory
current_dir = os.getcwd()
print(f"Current directory: {current_dir}")

# Look for WordPress files
wp_dirs = [
    "C:\\Users\\tjd20.LAPTOP-PCMC2SUO\\local-wordpress",
    "C:\\Users\\tjd20.LAPTOP-PCMC2SUO\\wordpress",
    "C:\\Users\\tjd20.LAPTOP-PCMC2SUO\\wordpress-local-test",
    "."
]

# Find a directory with WordPress files
wp_dir = None
for directory in wp_dirs:
    if os.path.exists(directory):
        # Check if it contains WordPress files
        if os.path.exists(os.path.join(directory, "wp-config.php")) or \
           os.path.exists(os.path.join(directory, "wp-content")) or \
           os.path.exists(os.path.join(directory, "wp-includes")):
            wp_dir = directory
            break

if wp_dir:
    print(f"Found WordPress files in: {wp_dir}")
    os.chdir(wp_dir)
else:
    print("No WordPress installation found. Using current directory.")

PORT = 8000

class MyHttpRequestHandler(http.server.SimpleHTTPRequestHandler):
    def do_GET(self):
        # Handle WordPress-style URLs
        if self.path.endswith('/'):
            # Check for index files
            if os.path.exists('index.php'):
                self.path = '/index.php'
            elif os.path.exists('index.html'):
                self.path = '/index.html'
        
        # If no file extension, try .php first for WordPress
        if not '.' in self.path.split('/')[-1] and not self.path.endswith('/'):
            if os.path.exists(self.path + '.php'):
                self.path += '.php'
        
        return http.server.SimpleHTTPRequestHandler.do_GET(self)

Handler = MyHttpRequestHandler

try:
    with socketserver.TCPServer(("", PORT), Handler) as httpd:
        print(f"Server started at http://localhost:{PORT}")
        print("Press Ctrl+C to stop the server")
        
        # Open browser in a separate thread
        def open_browser():
            time.sleep(1)
            webbrowser.open(f'http://localhost:{PORT}')
        
        browser_thread = threading.Thread(target=open_browser)
        browser_thread.daemon = True
        browser_thread.start()
        
        # Start the server
        httpd.serve_forever()
        
except KeyboardInterrupt:
    print("\nServer stopped.")
except OSError as e:
    if "Address already in use" in str(e):
        print(f"Port {PORT} is already in use. Try a different port:")
        print(f"python simple-server.py 8001")
    else:
        print(f"Error starting server: {e}")
except Exception as e:
    print(f"Error: {e}")